### ISSUE TITLE
New to Min

### ISSUE THREAD

**Issue by 7f043** [minbrowser/min#1171]
I found this browser just tonight, I'm already impressed. There are a handful of things I look for in a browser.

**Things I look for:** Simple, privacy focused, ad-blocking, with some kind of "vim" keyboard extension. This browser manages to hit the nail on the head. Well done. Believe it or not, it has been impossible to find all of these things in one place.

**Things I have yet to find/should be added/the browser is missing:** 

1. Default zoom for pages in settings. I find the default zoom for most webpages to be too small. (I can read it, but why strain the eyes like that?) I always tend to increase it to 133% or so.

2. the "new tab" page is too bright. This page should be dark when using the dark theme for the browser.

3. There should be an option to have "private tabs" be in a separate window. I appreciate this browser tries to do something new, but this is a massive problem when having 10+ tabs open. It can be disorienting.

4. A way to add css for specific sites. I generally hate the way certain sites are layed out, and I apply my own CSS to them. I like to remove elements that are otherwise ugly to me. It would be cool if I could do that here as well.

**Bugs:**

5. the VIM extension may have an issue where pressing "f" on a youtube video page will both simultaneously fullscreen the video, and also cause the link selector to appear

6. Adblocker doesn't seem to block pre-video ads on youtube.

Good job, I'm enjoying my new browser so far! Just a few suggestions for a hard-working team. Thanks for the consideration!

**Comment by Syndamia** [comment]
Hello @7f043 and thank you for your input! I'll try to answer some of your concerns.

> Default zoom for pages in settings. I find the default zoom for most webpages to be too small. (I can read it, but why strain the eyes like that?) I always tend to increase it to 133% or so.

Yes, we are aware. Other(s) have pointed this out before (#898).

> A way to add css for specific sites. I generally hate the way certain sites are layed out, and I apply my own CSS to them. I like to remove elements that are otherwise ugly to me. It would be cool if I could do that here as well.

You actually can! Through usercss (*it's best to first read [this wiki page](https://github.com/minbrowser/min/wiki/userscripts)*) you can create a script that adds a `style` element to a page with the desired CSS. [This](https://github.com/PalmerAL/min-dark-mode-userscript/blob/master/dark-mode.js) is a very good example of just that.

Also, to limit it to a certain website, you'll need to modify the `@match` parameter. You'll need to put `*` for the parts that can be whatever. For example: execute on **every** Google website `@match *google.com*`, this means that it doesn't matter what comes before `google.com` and what comes after it, there just needs to be the `google.com`. 

**That is a very bad example**, because you can have `google.com` anywhere in the link (*which is problematic if you, for example, search for `google.com`*) and also it will execute on pages like `mail.google.com`. It's best to make it more specific, so something like `@match *://www.google.com` or `@match *://mail.google.com`.

> the VIM extension may have an issue where pressing "f" on a youtube video page will both simultaneously fullscreen the video, and also cause the link selector to appear

This isn't that much a bug than a configuraation problem. `YouTube` has some [build-in shortcuts](https://duckduckgo.com/?q=youtube+shortcuts&t=min&ia=cheatsheet&iai=r1-0&page=1&adx=sltb&sexp=%7B%22v7exp%22%3A%22a%22%2C%22sltexp%22%3A%22b%22%2C%22wiadrk%22%3A%22b%22%7D), so the problem is that yout `Vim` shortcut executes, but also `YouTube` executes it's own shortcut. The solution is to either change the `Vim` shortcut or create a usercss file that blocks/changes `YouTube`'s shortcut (*I can't help with that, because I don't know how `YouTube` handles shortcuts*).

> Adblocker doesn't seem to block pre-video ads on youtube.

Yes, this is also a known problem (*first mentioned in #1121*). For now the only fix is to use [this usercss file](https://github.com/minbrowser/min/issues/1121#issuecomment-663021909). For me atleast it blocked the pre-video ads every time.

For `2` and `3` @PalmerAL should say.

**Comment by 7f043** [comment]
Thanks for the quick reply! Only one thing I would like to add,

> This isn't that much a bug than a configuraation problem. `YouTube` has some [build-in shortcuts](https://duckduckgo.com/?q=youtube+shortcuts&t=min&ia=cheatsheet&iai=r1-0&page=1&adx=sltb&sexp=%7B%22v7exp%22%3A%22a%22%2C%22sltexp%22%3A%22b%22%2C%22wiadrk%22%3A%22b%22%7D), so the problem is that yout `Vim` shortcut executes, but also `YouTube` executes it's own shortcut. The solution is to either change the `Vim` shortcut or create a usercss file that blocks/changes `YouTube`'s shortcut (_I can't help with that, because I don't know how `YouTube` handles shortcuts_).

I'm aware of this, but other browsers handle this by only allowing the youtube shortcuts to function after pressing "i" for insert mode which also blocks VIM command keybindings. So this issue doesn't happen. I wouldn't know how to fix this either, so I guess I'll look into it

**Comment by PalmerAL** [comment]
I agree a default zoom setting would be a good idea; I'd take a PR for that if anyone's interested in doing it.

> the "new tab" page is too bright. This page should be dark when using the dark theme for the browser.

Created #1173.

> There should be an option to have "private tabs" be in a separate window. I appreciate this browser tries to do something new, but this is a massive problem when having 10+ tabs open. It can be disorienting.

#169. Basically, supporting multiple windows is a lot of work, and I don't have enough time to do it right now, so it's probably not going to happen unless someone's interested in working on it.

> the VIM extension may have an issue where pressing "f" on a youtube video page will both simultaneously fullscreen the video, and also cause the link selector to appear

One easy workaround for this would be to change "f" to something that YT doesn't use as a shortcut; that's on line 203 of the script.

> Adblocker doesn't seem to block pre-video ads on youtube.

Switching the content blocking setting to "block all ads" blocks the Youtube video ads for me (as opposed to the ads on the homepage, which you need a userscript for). Maybe you could try that?

**Comment by 7f043** [comment]
Thanks for the responses.

Another good workaround is creating an "insert" mode. Allowing site shortcuts to be separate from vim ones, which wouldn't interfere with other websites too. I'm new to scripting, if you have any suggestions

 Do you have any suggestions for creating a new window? I wouldn't mind helping create something like this, but my experience is Minimal. :) no pun intended.

If I get any other responses, I'll start closing out this thread
